### PR TITLE
agent: enable enable_central_service_config by default

### PR DESCRIPTION
**Pull request by rboyer**

### CODE REVIEW

**Review by mkeeler — COMMENTED** [review]
Does the agent gracefully handle the case where the servers do not have connect enabled?

**Comment by rboyer** [comment]
(the branch is not done yet, i just figured I'd steal a little CI time to run tests for me using draft mode)

IIRC the linkage of connect being enabled and `enable_central_service_config=true` isn't direct. When the flag is enabled it means the ServiceManager attempts to use `proxy-defaults` and `service-defaults` during service registration for the service and it's upstreams. In the non-connect case someone just won't have any upstreams (that matter) and really the only thing that would be merged would be its own protocol value.

**Comment by rboyer** [comment]
Hmm, `TestAgent_ReloadConfigAndKeepChecksStatus` fails when this flag is set because on reload a check comes back `critical` instead of `passing`

File: agent/agent.go [hunk 1]
```diff
@@ -1859,12 +1860,13 @@ func (a *Agent) AddService(service *structs.NodeService, chkTypes []*structs.Che
...
+	})
 }
 
 // addServiceLocked adds a service entry to the service manager if enabled, or directly
 // to the local state if it is not. This function assumes the state lock is already held.
-func (a *Agent) addServiceLocked(req *addServiceRequest, snap map[structs.CheckID]*structs.HealthCheck) error {
+func (a *Agent) addServiceLocked(req *addServiceRequest) error {
 	req.fixupForAddServiceLocked()
 
 	req.service.EnterpriseMeta.Normalize()
```
Comment (rboyer) [inline agent/agent.go:1872]: Below on the line just after `if a.config.EnableCentralServiceConfig {` (that I can't comment on) is where `req` loses `snap` when we shunt this function call through service manager. The fix is to make sure `snap` is attached to `req`.

**Review by freddygv — APPROVED** [review]
LGTM, just missing a changelog entry

**Pull request merged**